h2o-algos/src/main/java/hex/tree/xgboost/XGBoostGPULock.java

```diff
@@ -0,0 +1,21 @@
+package hex.tree.xgboost;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class XGBoostGPULock {
```

Comment:
Good points, will fix!

Other thing is the conversion methods from frame/chunks -> DMatrix are pretty similar so I want to somehow DRY them but there are enough small differences that it's a bit annoying to do I think.